gossip/privdata/distributor.go

```diff
@@ -227,8 +227,17 @@ func (d *distributorImpl) disseminationPlanForMsg(colAP privdata.CollectionAcces
 	m := pvtDataMsg.GetPrivateData().Payload
 
 	eligiblePeers := d.eligiblePeersOfChannel(routingFilter)
-	identitySets := d.identitiesOfEligiblePeers(eligiblePeers, colAP)
 
+	// With the shift to per peer dissemination in FAB-15389, we must first check
+	// that there are enough eligible peers to satisfy RequiredPeerCount.
+	if (len(eligiblePeers)) < colAP.RequiredPeerCount() {
+		return nil, fmt.Errorf("required to disseminate to at least %d peers, but know of only %d eligible peers", colAP.RequiredPeerCount(), len(eligiblePeers))
```

Comment:
We seem to have a good set of methods to ensure that pvtData is never lost and also the block commit performance is not impacted. 

1. PvtData should be disseminated to the `requiredPeerCount` number of peers. Otherwise, the endorsement is rejected (fixed by this PR).
2. As per the collection's endorsement policy, by default, the PvtData is replicated across the transient stores.
3. If a peer does not find PvtData in the transient store during the commit time, it would fetch from other peers. If the peer could not fetch the required pvtData from any other peers, it would record that particular pvtData is missing and proceed with the commit. Later, the reconciler fetches the missing pvtData from other peers. 
4. Most BaaS platforms also provide HA  — one from IBP https://cloud.ibm.com/docs/blockchain-sw?topic=blockchain-sw-ibp-console-ha 

As there could be duplicate communications (i.e., every endorser disseminating the pvt data even to the peer which already has the pvtData -- happens when there are multiple endorsers), such an approach in WAN could be bad for the performance. Anyway, this needs to be benchmarked to see whether it is really a problem. If we find this to be a problem, we can keep track of the previous senders and reduce the number of dissemination (for the future).

Reply:
I think the onus is on the admin to configure `requiredPeerCount` and `maxPeerCount` depending on the endorsement policy to avoid duplicate transmissions.

Reply:
Makes sense. I'm looking forward to your private data perf test results.